crates/kornia-apriltag/src/lib.rs

```diff
@@ -27,3 +45,455 @@ pub mod quad;
 
 /// Decoding utilities for AprilTag detection.
 pub mod decoder;
+
+#[derive(Debug, Clone, PartialEq)]
+/// Configuration for decoding AprilTags.
+pub struct DecodeTagsConfig {
+    /// List of tag families to detect.
+    pub tag_families: Vec<TagFamily>,
+    /// Configuration for quad fitting.
+    pub fit_quad_config: FitQuadConfig,
+    /// Whether to enable edge refinement before decoding.
+    pub refine_edges_enabled: bool,
+    /// Sharpening factor applied during decoding.
+    pub decode_sharpening: f32,
+    /// Whether normal border tags are present.
+    pub normal_border: bool,
+    /// Whether reversed border tags are present.
+    pub reversed_border: bool,
+    /// Minimum tag width at border among all families.
+    pub min_tag_width: usize,
+    /// Minimum difference between white and black pixels for thresholding.
+    pub min_white_black_difference: u8,
+}
+
+impl DecodeTagsConfig {
+    /// Creates a new `DecodeTagsConfig` with the given tag families.
+    pub fn new(tag_families: Vec<TagFamily>) -> Self {
+        let mut normal_border = false;
+        let mut reversed_border = false;
+        let mut min_tag_width = usize::MAX;
+
+        tag_families.iter().for_each(|family| {
+            if family.width_at_border < min_tag_width {
+                min_tag_width = family.width_at_border;
+            }
+            normal_border |= !family.reversed_border;
+            reversed_border |= family.reversed_border;
+        });
+
+        if min_tag_width < 3 {
+            min_tag_width = 3;
+        }
+
+        // TODO: Set the default values which are best for the given tags
+        Self {
+            tag_families,
+            fit_quad_config: Default::default(),
+            normal_border,
+            refine_edges_enabled: true,
+            decode_sharpening: 0.15,
+            reversed_border,
+            min_tag_width,
+            min_white_black_difference: 20,
+        }
+    }
+
+    /// Creates a `DecodeTagsConfig` with all supported tag families.
+    pub fn all() -> Self {
+        let tag_families = vec![
+            TagFamily::tag16_h5(),
```

Comment:
maybe the all() method should be implemented at the TagFamily level to easily keep it updated